### DISCUSSION TITLE
How to get the 'instrument_name' out of radar.metadata

### DISCUSSION THREAD

**Discussion by salbenson** [ARM-DOE/pyart#1650] [Q&A] [answered]
Hi, 
I am trying to pull the instrument_name from the radar object metadata, and I can't figure out how to do that.  I am looking to grab the string 'SPOL' from this metadata.  Here is my metadata:
>>> print(radar.metadata)
{'Conventions': 'CF-1.7', 'Sub_conventions': 'CF-Radial instrument_parameters radar_parameters radar_calibration', 'version': 'CF-Radial-1.4', 'title': '', 'institution': '', 'references': '', 'source': '', 'history': '', 'comment': '', 'original_format': 'CFRADIAL', 'driver': 'RadxConvert(NCAR)', 'created': '2023/02/17 01:13:41.224', 'start_datetime': '2022-06-29T08:06:59Z', 'time_coverage_start': '2022-06-29T08:06:59Z', 'start_time': '2022-06-29 08:06:59.851', 'end_datetime': '2022-06-29T08:09:16Z', 'time_coverage_end': '2022-06-29T08:09:16Z', 'end_time': '2022-06-29 08:09:16.773', 'instrument_name': 'SPOL', 'site_name': 'Nanliao', 'scan_name': 'PrecipRhi2', 'scan_id': 0, 'platform_is_mobile': 'false', 'n_gates_vary': 'false', 'ray_times_increase': 'true', 'volume_number': 722, 'platform_type': 'fixed', 'instrument_type': 'radar', 'primary_axis': 'axis_z'}

Thank you for your help, 
Sally

**Comment by zssherman** [comment]
Hi @salbenson! For that, can you try:
```
print(radar.metadata['instrument_name'])
```
That should then print out the 'SPOL'

**Reply by zssherman** [reply]
If not there then I would try:
```
print(radar.instrument_parameters['instrument_name'])

**Reply by salbenson** [reply]
Thanks for your help.  When I type that at the python command line I get:
print(radar.instrument_parameters['instrument_name'])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
KeyError: 'instrument_name'

I am not sure what I am doing wrong.

**Reply by salbenson** [reply]
I am getting the same thing for print(radar.instrument_parameters['instrument_name'])
>>> print(radar.instrument_parameters['instrument_name'])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
KeyError: 'instrument_name'

**Reply by zssherman** [reply]
@salbenson Apologies I edited my response a little late. Can you swap out instrument_parameters with metadata instead?

```
radar.metadata['instrument_name']
```

**Reply by salbenson** [reply, marked as answer]
It was
>>> print(radar.metadata['instrument_name'])
SPOL
that worked.  Thank you so much.  I am kind of new to python.  I am a longtime IDL and FORTRAN programmer.

**Reply by zssherman** [reply]
No worries! Py-ART stores alot of that metadata in python dictionaries. For most of it you can index it with brackets. Like units can be found in a field with:
```
radar.fields['reflectivity']['units']
```
etc
